src/include/server/mir/observer_multiplexer.h

```diff
@@ -264,28 +268,36 @@ void ObserverMultiplexer<Observer>::register_interest(
     observers.emplace_back(std::make_shared<WeakObserver>(observer, executor));
 }
 
+template<class Observer>
+void ObserverMultiplexer<Observer>::register_early_observer(
+    std::weak_ptr<Observer> const& observer,
+    Executor& executor)
+{
+    std::lock_guard lock{observer_mutex};
+
+    early_observers.emplace_back(std::make_shared<WeakObserver>(observer, executor));
```

Comment:
I think there could be a significant simplification (eliminating `early_observers`) by using:

```suggestion
    observers.push_front(std::make_shared<WeakObserver>(observer, executor));
```

(With a note on `observers` that partitioning early observers before other observers is an invariant.)

Reply:
Yeah, I've just done that. Will include the note about the invariant.

Reply:
Done.